pkg/provider/kubernetes/ingress/kubernetes.go

```diff
@@ -599,6 +614,46 @@ func (p *Provider) loadService(client Client, namespace string, backend netv1.In
 
 			return svc, nil
 		}
+
+		if svcConfig.Service.NodePortLB && service.Spec.Type == corev1.ServiceTypeNodePort {
+			nodes, nodesExists, nodesErr := client.GetNodes()
+			if nodesErr != nil {
+				return nil, nodesErr
+			}
+
+			if !nodesExists || len(nodes) == 0 {
+				return nil, fmt.Errorf("nodes not found in namespace %s", namespace)
+			}
+
+			protocol := getProtocol(portSpec, portSpec.Name, svcConfig)
+
+			port, err := getNodePort(service)
```

Comment:
Shouldn't you reuse the NodePort of the port selected previously (portSpec) or am I missing something?

Reply:
You are absolutely right. getNodePort is not needed here. I guess that I copied the pattern from the other provider.